nntrainer/include/layer.h

```diff
@@ -119,27 +119,20 @@ typedef enum {
   WEIGHT_UNKNOWN
 } WeightInitializer;
 
-/**
- * @brief   Print Options when printing layer info
- */
-typedef enum {
-  // clang-format off
-  PRINT_INST_INFO  = (1 << 0), /**< Option to print type & instance address info */
-  PRINT_SHAPE_INFO = (1 << 1), /**< Option to print shape information, invalid before initiation*/
-  PRINT_PROP       = (1 << 2), /**< Option to print properties */
-  PRINT_PROP_META  = (1 << 3), /**< Option to print properties that describe meta info
-                                    e.g) layer activation type for non-activation layer. */
-  PRINT_WEIGHTS    = (1 << 4), /**< Option to print weights */
-  PRINT_METRIC     = (1 << 5)  /**< Option to print metrics (currently loss only) */
-  // clang-format on
-} LayerPrintOption;
-
 /**
  * @class   Layer Base class for layers
  * @brief   Base class for all layers
  */
 class Layer {
 public:
+  enum class PrintPreset {
```

Comment:
I think this enum might only need PRINT_SUMMARY*? Differentiation between PRINT_SUMMARY* and PRINT_ALL could be better done with `ML_TRAIN_SUMMARY_LAYER` and `ML_TRAIN_SUMMARY_TENSOR`.

Reply:
>Differentiation between PRINT_SUMMARY* and PRINT_ALL could be better done with ML_TRAIN_SUMMARY_LAYER and ML_TRAIN_SUMMARY_TENSOR.
 
This can be done. However, I also need differentiation between PRINT_SUMMARY, PRINT_SUMMARY_META which is depending on the `initialized` member.

Similar things can happen for more delicate presets. So I think it is better that the model chooses what to print depending on its state since the layer does not know about the model's state or more, it can be printed without a model owns the layer.

Please let me know in detail if you have a better design. I might have missed some point

Reply:
Yes, model should help decide the summary level for layer (differentiate between SUMMARY and SUMMARY_META). But the decision of TENSOR should not be taken by model, but rather than trickle down to layer.